The failure is
```
 running: "/nix/store/4akw6n9a2f1cxwcwihzf1diblbyzhkkc-clang-wrapper-7.1.0/bin/cc" "-O3" "-ffunction-sect
  cargo:warning=In file included from xz-5.2/src/liblzma/common/filter_decoder.c:13:
  cargo:warning=In file included from xz-5.2/src/liblzma/common/filter_decoder.h:16:
  cargo:warning=In file included from xz-5.2/src/liblzma/common/common.h:16:
  cargo:warning=xz-5.2/src/common/sysdefs.h:24:11: fatal error: 'config.h' file not found
```

--- a/pkgs/development/tools/rust/rustup/default.nix
+++ b/pkgs/development/tools/rust/rustup/default.nix
@@ -10,6 +10,8 @@
 , zlib
 , Security
 , CoreServices
+, libiconv
+, xz
 }:
 
 let
@@ -36,7 +38,7 @@ rustPlatform.buildRustPackage rec {
   buildInputs = [
     curl
     zlib
-  ] ++ lib.optionals stdenv.isDarwin [ CoreServices Security ];
+  ] ++ lib.optionals stdenv.isDarwin [ CoreServices Security libiconv xz ];
 
   cargoBuildFlags = [ "--features no-self-update" ];
 

